lib/puppet/pops/lookup/configured_data_provider.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'lookup_config'
+require_relative 'data_provider'
+require 'puppet/data_providers/hiera_config'
+
+module Puppet::Pops
+module Lookup
+# @api private
+class ConfiguredDataProvider
```

Comment:
Is this essentially "AbstractV4DataProvider" (new base class for the ones being deprecated), or the new baseclass for both the old and new (V5) function providers? I find the name confusing - "ConfiguredDataProvider" as opposed to what "NonConfigured" ? If this is a temporary thing (for v4) then nevermind.